[lodepng] Update to commit e34ac0@2020-03-15

BurningEnlightenment opened the pull request:
I'm currently preparing a cmake integrated bgfx port (requested here #1303) which requires a newer lodepng version. Additionally I added proper cmake import configs, because otherwise the debug library build cannot be found (the release binary always takes precedence).

BurningEnlightenment committed:
[lodepng] Add cmake import configs

BurningEnlightenment committed:
[lodepng] Update to commit e34ac0@2020-03-15

Cheney-W commented on ports/lodepng/portfile.cmake:
```diff
@@ -25,7 +24,12 @@ vcpkg_configure_cmake(
 vcpkg_copy_tool_dependencies(${CURRENT_PACKAGES_DIR}/tools/lodepng)
 
 
 file(INSTALL ${SOURCE_PATH}/lodepng.h DESTINATION ${CURRENT_PACKAGES_DIR}/share/lodepng RENAME copyright)
```
Could you please modify /share/lodepng as /share/${PORT} ? Thank you.

Cheney-W commented on ports/lodepng/portfile.cmake:
```diff
@@ -25,7 +24,12 @@ vcpkg_configure_cmake(
 
 file(INSTALL ${SOURCE_PATH}/lodepng.h DESTINATION ${CURRENT_PACKAGES_DIR}/share/lodepng RENAME copyright)
+
+vcpkg_test_cmake(PACKAGE_NAME lodepng)
```
The vcpkg_test_cmake() function has been removed from VCPKG, so could you please remove this sentence from your change? Thank you.

BurningEnlightenment replied:
Done. However, I'd like to point out that `.\vcpkg create` still adds that function call to all newly generated projects.

Cheney-W replied:
Thanks for pointing it out. We will consider updating this feature.

BurningEnlightenment committed:
[lodepng] Modernize port

- Remove calls to deprecated vcpkg functions.
- Remove commented out pngdetail tool project definitions.

ras0219-msft commented:
/azp run

BurningEnlightenment committed:
[lodepng] Remove osx failure from ci.baseline

dan-shaw commented:
/azp run
